### PR TITLE
react hooks: add base interfaces Listenable, StatusParmas and ChatStatusResponse

### PR DESCRIPTION
**Pull request by vladvelici**
### Context

Adds interfaces as they are defined in CHADR-030. Closes [CHA-383]. They will be used for future hooks.

### Checklist

* [x] QA'd by the author.
* [ ] Unit tests created (if applicable).
* [ ] Integration tests created (if applicable).
* [x] Follow coding style guidelines found [here](https://github.com/ably/engineering/tree/main/best-practices).
* [ ] TypeDoc updated (if applicable).
* [ ] (Optional) Update documentation for new features.
* [ ] Browser tests created (if applicable).
* [ ] In repo demo app updated (if applicable).

### Testing Instructions (Optional)

n/a

[CHA-383]: https://ably.atlassian.net/browse/CHA-383?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by AndyTWF — COMMENTED** [review]
All of these interfaces will be public facing (`Listenable` by proxy via classes that implement it), so:

- Can we add `StatusParams` and `ChatStatusResponse` to `index.ts`
- Can we add typedoc comments

**Comment by vladvelici** [comment]
@AndyTWF done both

File: src/react/chat-status-response.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { ConnectionStatus, ErrorInfo, RoomStatus } from '@ably/chat';
+
+/**
+ * Common status variables for chat features. Most hooks in this library implement this interface.
```
Comment (splindsay-92) [inline src/react/chat-status-response.ts:4]: To make typedocs work, you will need to adjust the paths typedocs uses to find the exported types. My open PR does this if you want to copy it :) Else it should be merged soon and a rebase will fix this!

**Review by AndyTWF — APPROVED** [review]

**Comment by AndyTWF** [comment]
Approved (pending the fix in Stevens PR and rebase)

**Review by splindsay-92 — APPROVED** [review]
Approving in lieu of the typedocs fix.

**Comment by vladvelici** [comment]
Does anyone know why the test coverage fails here? There are no new functions added, just interfaces and their imports.

cc @AndyTWF @splindsay-92

**Comment by AndyTWF** [comment]
> Does anyone know why the test coverage fails here? There are no new functions added, just interfaces and their imports.
> 
> cc @AndyTWF @splindsay-92

I've noticed that for whatever reason, v8 isn't detecting that some files are just declarations. We might need to just adjust the thresholds (as soon as we start adding more hooks they'll go back up again)

**Pull request merged**